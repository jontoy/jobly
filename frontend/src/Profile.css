.Profile form {
    text-align: left;
}