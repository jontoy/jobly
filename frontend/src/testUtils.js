import React from "react";
import UserContext from "./userContext";

const demoUser = {
  username: "test",
  first_name: "test",
  last_name: "mctest",
  email: "test@test.net",
  photo_url: null,
  jobs:[]
};

const UserProvider = ({ children, currentUser = demoUser, appliedIds = [] }) => (
  <UserContext.Provider value={{currentUser, appliedIds}}>
    {children}
  </UserContext.Provider>
);

export { UserProvider };
