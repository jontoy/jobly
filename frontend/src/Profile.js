import React, {useContext, useState} from 'react';
import {Link} from 'react-router-dom';
import UserContext from './userContext';
import JoblyApi from './JoblyApi';
import './Profile.css';

const Profile = () => {
    const {currentUser, setCurrentUser} = useContext(UserContext);
    const {first_name, last_name, email} = currentUser;
    const [formData, setFormData] = useState({first_name, last_name, email, password:''})
    const handleChange = (e) => {
        const {name, value} = e.target;
        setFormData(data => ({
            ...data,
            [name]: value
        }));
    }
    const handleSubmit = async (e) => {
        e.preventDefault();
        let updatedUser = await JoblyApi.updateUser(currentUser.username, formData);
        setCurrentUser(currUser => ({
            ...currUser,
            user: updatedUser
        }))
    } 
    return (
        <div className="Profile container mt-3">
            <div className="card pt-2">
                <h4 className="card-title">{currentUser.username}'s Profile:</h4>
                <div className="card-body">
                    <form onSubmit={handleSubmit}>
                        <div className="form-group">
                            <label htmlFor="first_name">First Name</label>
                            <input 
                                className="form-control"
                                type="text" 
                                value={formData.first_name} 
                                id="first_name"
                                name="first_name" 
                                placeholder="e.g. John" 
                                onChange={handleChange}/>
                        </div>
                        <div className="form-group">
                            <label htmlFor="last_name">Last Name</label>
                            <input 
                                className="form-control"
                                type="text" 
                                value={formData.last_name} 
                                id="last_name"
                                name="last_name" 
                                placeholder="e.g. Smith" 
                                onChange={handleChange}/>
                        </div>
                        <div className="form-group">
                            <label htmlFor="email">Email Address</label>
                            <input 
                                className="form-control"
                                type="text" 
                                value={formData.email} 
                                id="email"
                                name="email" 
                                placeholder="e.g. johnsmith@gmail.com" 
                                onChange={handleChange}/>
                        </div>
                        <div className="form-group">
                            <label htmlFor="password">Confirm with Password</label>
                            <input 
                                className="form-control"
                                type="password" 
                                value={formData.password} 
                                id="password"
                                name="password" 
                                placeholder="Password" 
                                onChange={handleChange}/>
                        </div>
                        <button className="btn btn-success">Save Updated Information</button>
                    </form>
                </div>
                <div className="card-footer">
                    <Link to="/applications"><button className="btn btn-outline-info">View current applications</button></Link>
                </div>
            </div>
        </div>
    )
}

export default Profile;