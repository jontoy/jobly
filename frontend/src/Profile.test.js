import React from 'react';
import { render } from '@testing-library/react';
import Profile from './Profile';
import { MemoryRouter } from 'react-router-dom';
import {UserProvider} from './testUtils';

it('renders without crashing', () => {
    render(
    <MemoryRouter>
        <UserProvider>
            <Profile />
        </UserProvider>
    </MemoryRouter>
    );
});
  
it('matches snapshot when user is logged in', () => {
      const {asFragment} = render(
        <MemoryRouter>
            <UserProvider>
                <Profile />
            </UserProvider>
        </MemoryRouter>
        );
    expect(asFragment()).toMatchSnapshot();
});