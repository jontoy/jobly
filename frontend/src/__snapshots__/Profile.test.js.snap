// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot when user is logged in 1`] = `
<DocumentFragment>
  <div
    class="Profile container mt-3"
  >
    <div
      class="card pt-2"
    >
      <h4
        class="card-title"
      >
        test's Profile:
      </h4>
      <div
        class="card-body"
      >
        <form>
          <div
            class="form-group"
          >
            <label
              for="first_name"
            >
              First Name
            </label>
            <input
              class="form-control"
              id="first_name"
              name="first_name"
              placeholder="e.g. John"
              type="text"
              value="test"
            />
          </div>
          <div
            class="form-group"
          >
            <label
              for="last_name"
            >
              Last Name
            </label>
            <input
              class="form-control"
              id="last_name"
              name="last_name"
              placeholder="e.g. Smith"
              type="text"
              value="mctest"
            />
          </div>
          <div
            class="form-group"
          >
            <label
              for="email"
            >
              Email Address
            </label>
            <input
              class="form-control"
              id="email"
              name="email"
              placeholder="e.g. johnsmith@gmail.com"
              type="text"
              value="test@test.net"
            />
          </div>
          <div
            class="form-group"
          >
            <label
              for="password"
            >
              Confirm with Password
            </label>
            <input
              class="form-control"
              id="password"
              name="password"
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
          <button
            class="btn btn-success"
          >
            Save Updated Information
          </button>
        </form>
      </div>
      <div
        class="card-footer"
      >
        <a
          href="/applications"
        >
          <button
            class="btn btn-outline-info"
          >
            View current applications
          </button>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
